src/kani-compiler/rustc_codegen_kani/src/overrides/hooks.rs

```diff
@@ -146,13 +146,51 @@ impl<'tcx> GotocHook<'tcx> for Assume {
     }
 }
 
+struct Assert;
+impl<'tcx> GotocHook<'tcx> for Assert {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        matches_function(tcx, instance, "KaniAssert")
+    }
+
+    fn handle(
+        &self,
+        tcx: &mut GotocCtx<'tcx>,
+        _instance: Instance<'tcx>,
+        mut fargs: Vec<Expr>,
+        _assign_to: Option<Place<'tcx>>,
+        target: Option<BasicBlock>,
+        span: Option<Span>,
+    ) -> Stmt {
+        assert_eq!(fargs.len(), 2);
+        let cond = fargs.remove(0).cast_to(Type::bool());
+        let msg = fargs.remove(0);
+        let msg = utils::extract_const_message(&msg).unwrap();
+        let target = target.unwrap();
+        let loc = tcx.codegen_span_option(span);
```

Comment:
Have you checked how the trace looks like in viewer? I'm wondering if this will look weird since we are using two different locations.

Reply:
The trace does look a bit weird in viewer (notice the missing files/lines):
```
Step 38: Function foo, File reach.rs, Line 4
assert!(x < 4);
var_5 = FALSE 
Step 39: Function MISSING, File MISSING, Line 0
temp_0 = FALSE 
Step 40: Function MISSING, File MISSING, Line 0
temp_0 = FALSE 
Step 41: Function foo, File reach.rs, Line 4
assert!(x < 4);
failure: foo.assertion.1: assertion: x < 4
```
I'll see if changing all the locations to the "caller" location fixes it.

Reply:
Cool. BTW, the `MISSING` is a different problem. We should chat with @markrtuttle about this. Viewer doesn't seem to handle files that are not inside the current folder.